week-1-ts-ecosystem/w01e05-my-own-cli/src/demo-prepare/actions/autoGenerateProductsAction.ts

```diff
@@ -0,0 +1,50 @@
+import { chooseProductTypeAction } from './choseProductTypeAction';
+import prompts from 'prompts';
+import chalk from 'chalk';
+import { v4 } from 'uuid';
+import { faker } from '@faker-js/faker';
+import { PRODUCT_TYPE } from 'w01e03-object-oriented-structure/src/models/product/Product';
+import { MAX_NUMBER_OF_PRODUCTS } from './consts';
+import { Data } from './utils';
+import { DataItem } from './types';
+
+export const autoGenerateProductsAction = async (): Promise<void> => {
+  const productType = await chooseProductTypeAction();
+  let isNumberOfProductsValid = false;
+  let numberOfProducts = 1;
+
+  while (!isNumberOfProductsValid) {
+    const { numberOfProducts: givenNumberOfProducts } = await prompts({
+      name: 'numberOfProducts',
+      type: 'number',
+      message: 'Podaj ile produktów wygenerować (max 8):',
+    });
+
+    if (
```

Comment:
Wiesz co: https://www.npmjs.com/package/prompts

ma opcję `validate` - w której wydaje mi się że mógłbyś to zamknąć. 
(bo tam się funkcję podaje)

Wtedy nie potrzebowałbyś tego `while` - sprawdź.